### PR TITLE
Use calculate_residuals for SSPRK and Heun/Ralston

### PR DESCRIPTION
**Pull request by ranocha**
Cf. #511.

### CODE REVIEW

File: src/perform_step/ssprk_perform_step.jl [hunk 1]
```diff
@@ -514,7 +516,8 @@ end
   step_limiter!(u, f, t+dt)
 
   if integrator.opts.adaptive
-    @. atmp = (utilde-u)/(integrator.opts.abstol+max(integrator.opts.internalnorm(uprev),integrator.opts.internalnorm(u))*integrator.opts.reltol)
+    @. utilde = utilde - u
```
Comment (YingboMa) [inline src/perform_step/ssprk_perform_step.jl]: Is it possible to transform the coefficients such that we only need to perform one linear combination, but not to subtract two arrays? e.g. https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/blob/master/src/perform_step/low_order_rk_perform_step.jl#L23

Comment (ranocha) [reply src/perform_step/ssprk_perform_step.jl]: I don't see how this could be possible without using more memory to save more temporary arrays.

Comment (YingboMa) [reply src/perform_step/ssprk_perform_step.jl]: Ah, I see. Thanks for explaining!

**Comment by ChrisRackauckas** [comment]
No, there are ways. For example

https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/blob/9a2823d3d88cce0438d6dcadb90eb0d21e2ea914/src/perform_step/ssprk_perform_step.jl#L462-L465

```julia
  u = u + dt_2*k
  if integrator.opts.adaptive
    utilde = (uprev + 2*u) / 3
  end
```

could instead be

```julia
  u = u + dt_2*k
  if integrator.opts.adaptive
    utilde = (uprev - u) / 3
  end
```

and now that's what used to be `utilde-u`. I wouldn't expect these transformations to have a large timing or numerical stability effect though, but it could be worth a further PR.

**Pull request merged**